spec/discovery.md

```diff
@@ -59,6 +59,18 @@ curl $(echo "$urltmpl" | sed -e "s/{name}/$name/" -e "s/{version}/$version/" -e
 
 where _name_, _version_, _os_, and _arch_ are set to their respective values for the image, and _ext_ is either `aci` or `aci.asc` for retrieving an App Container Image or signature respectively.
 
+The client MUST accept only best matching templates. Best matching templates are the one where all the templates labels can be substituted and with the highest number of substituted labels.
```

Comment:
> all the templates labels can be substituted and with the highest number of substituted labels

I don't really get the rationale here. If _all_ client-supplied labels _must_ be substituted, then won't be "highest number of substituted labels" always be equal to the number of client-supplied labels?

Or are you talking about templates where the same variable might exist multiple times? e.g. given client-supplied label `{name}`, then `{name}/{name}.{ext}` being somehow preferable to `{name}.{ext}` - if so, why?